pkg/codegen/docs/gen.go

```diff
@@ -1757,11 +1757,37 @@ func (mod *modContext) gen(fs codegen.Fs) error {
 	dctx := mod.docGenContext
 	modName := mod.getModuleFileName()
 
-	addFile := func(name, contents string) {
-		p := path.Join(modName, name, "_index.md")
+	seen := map[string]struct{}{}
+
+	addFile := func(name, contents string, prefixes []string) {
+		var namePrefixed string
+		for _, prefix := range prefixes {
+			namePrefixed = prefix + name
+			if _, exists := seen[namePrefixed]; exists {
+				// Unset namePrefixed because it isn't valid.
+				namePrefixed = ""
+				continue
+			}
+			break
+		}
+		if name != "" && namePrefixed == "" {
+			return
```

Comment:
Adding a hard failure on this would change the behavior for downstream users of this such as the registry. I'm not sure if we should modify the behavior in this way.

Reply:
If we can't fail here, at least a warning is merited.
Silently skipping files will just make these conflicts harder to debug for users.

Reply:
Since it's in codegen, I think it's okay if we fail in the registry. If that happens, the Docs team will open a fix or a P1, and we can resolve that before they upgrade.

A warning might also suffice here; does anyone review the logs during registrygen? I think the answer is no. If it were a manual task, then we'd guarantee eyes on the warning.

Again, just my 2c. Not my hill to die on. :)

Reply:
Yep, a failure would do too. A warning is slightly extra steps, but it's still more than silence.
Silently ignoring the failure is the only case I'd push back against.